ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -305,6 +314,7 @@ private PutOperation(RouterConfig routerConfig, NonBlockingRouterMetrics routerM
     isEncryptionEnabled = passedInBlobProperties.isEncrypted();
     restRequest = options.getRestRequest();
     loggingContext = makeLoggingContext();
+    compressionService = new CompressionService(routerConfig.getCompressionConfig(), routerMetrics.compressionMetrics);
```

Comment:
One for each operation?

Reply:
Not ideal.  We should use Dependency injection so we create only one.  Let me create a ticket to come back to refactor this to use dependency injection.https://jira01.corp.linkedin.com:8443/browse/AMBRY-9407